### ISSUE TITLE
[NPC][WotlK] Iron Dwarf Magus and Iron Dwarf Assailant Part 1, missing paths, mini event

### ISSUE THREAD

**Issue by Teridan-dal-wow** [dalaranwow/dalaran-wow#4322]
[//]: # (INFORMATION ABOUT TAGS)
[//]: # (The black and yellow tag is unique. There can only be one label assigned to them - one black and one yellow)
[//]: # (Priority - ... tags are assigned by team members)
[//]: # (Your issue will be reviewed by testers and the developers will fix it in the future.)
[//]: # (Only once the issue got the label [Fixed - On Live Server] you can find the fix on your Gameserver.)
[//]: # (You can increase the relevance of your issue by adding a thumb-up emote http://i.imgur.com/RFouVVi.png)

[//]: # (***************************************************************)
[//]: # (** !!!!!!!!!!!!!DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED!!!!!!!!!!!! **)
[//]: # (***************************************************************)

**Current behaviour**: Npc Iron Dwarf Magus http://wotlk.openwow.com/npc=29979 and Iron Dwarf Assailant http://wotlk.openwow.com/npc=29978 are missing their paths and there is a mini event in which these 2 npcs enters a Iron Dwarf Mole Machine http://wotlk.openwow.com/object=194316 and despawns. Area relevent for this report:
![wowscrnshot_012617_105642](https://cloud.githubusercontent.com/assets/6797611/22325329/31abdf16-e3b6-11e6-8931-3d160810d8e1.jpg)
![wowscrnshot_012617_105638](https://cloud.githubusercontent.com/assets/6797611/22325333/31b0adc0-e3b6-11e6-8b07-dbc41b2db565.jpg)
![wowscrnshot_012617_105631](https://cloud.githubusercontent.com/assets/6797611/22325330/31ae3b30-e3b6-11e6-92ea-a0246c64e5a2.jpg)
![wowscrnshot_012617_105621](https://cloud.githubusercontent.com/assets/6797611/22325331/31ae7c08-e3b6-11e6-87c9-962a7839e81c.jpg)
![wowscrnshot_012617_105649](https://cloud.githubusercontent.com/assets/6797611/22325332/31af54d4-e3b6-11e6-829e-8e1350700692.jpg)

**Expected behaviour**: Video from retail legion from 26.01.2017:
Part 1- https://www.youtube.com/watch?v=_j1JyvZ5mng
Part 2- https://www.youtube.com/watch?v=BVbOYgNEnFw
Part 3- https://www.youtube.com/watch?v=1SWx5B4QBQE
Part 4- https://www.youtube.com/watch?v=SiE1Ht7Rn7c

How should the event works (from what i saw):
 - There are 3 kind of groups made of Iron Dwarf Magus and Iron Dwarf Assailant:
1) Patroles - they do not take part of the event.
2) "Static" spawns - Iron Dwarf Magus and Iron Dwarf Assailant which are spawned on specific locations and do nothing (just stand still) - they do not take part of the event.
3) "Dynamic" spawns - Iron Dwarf Magus and Iron Dwarf Assailant which are spawned on specific locations and wait for a mole machine to spawn to run for it - they take part of the event.
 - Iron Dwarf Mole Machine spawns on specific locations every 60 secs (not all of them at the same time).
 - When a Iron Dwarf Mole Machine spawns, the nearest "dynamic" spawn of Iron Dwarf Magus or Iron Dwarf Assailant runs for it (just 1), if there isn't avaliable one, the machine just despawns.

**Comment by Rushor** [comment]
hm i think we need to script this c++ too, there too many variables, it cannot be done with SAI script

**Comment by Rushor** [comment]
or wait we can do it with the Datebase, but the eventmust be a bit adjusted for it

The molemachine will summon the Dwarf Magus or Iron Dwarf Assailant after the animation (it would look even better than on retail then)

**Comment by Teridan-dal-wow** [comment]
No no, the idea behind this event is that from that point all dwarfs launch their invasion at storm peaks. So you have point A (this event pointed above) and point B (all over storm peaks) like here:
![wowscrnshot_012617_132359](https://cloud.githubusercontent.com/assets/6797611/22329640/d6c56c38-e3ca-11e6-88c5-bc5f5429dfa6.jpg)
![wowscrnshot_012617_132351](https://cloud.githubusercontent.com/assets/6797611/22329639/d6c4b0f4-e3ca-11e6-9d49-c608a79b3be7.jpg)
![wowscrnshot_012617_132325](https://cloud.githubusercontent.com/assets/6797611/22329641/d6d4327c-e3ca-11e6-97ed-84ac77bd6496.jpg)
![wowscrnshot_012617_132316](https://cloud.githubusercontent.com/assets/6797611/22329643/d6d975f2-e3ca-11e6-949f-a85873ea77af.jpg)
![wowscrnshot_012617_132439](https://cloud.githubusercontent.com/assets/6797611/22329642/d6d626e0-e3ca-11e6-86e2-6915397a0d1a.jpg)

Thats why its really important to look exacly on retail.

**Comment by Rushor** [comment]
what do you mean?

i proposed to script it like that:

the molemachines that are static (the ones from your video in the area Terrace of the Makers) will summon the Dwarf Magus or Iron Dwarf Assailant after they did their spawnanimation. Then after the spawnanimation is done, the Dwarf Magus or Iron Dwarf Assailant will walk to the machine, will despawn and the molemachine will submerge again.

And every molemachine would summon it's own specific Dwarf Magus or Iron Dwarf Assailant in front of the machine.

The reason why it's hard to script this: in Core and DB be cannot split between the dynamic and the static spawns :) I hope i explained it better this time haha. my english is bad

**Comment by Teridan-dal-wow** [comment]
Ok, now i understand what you mean. Yes that way can work out too.

**Comment by Teridan-dal-wow** [comment]
I was thinking about how to improve the outcome of this issue and i can suggest this:
- if you know that at 11:00:00 time a mole mashine will spawn at "Point B"
- you could add a spawn location at "Point A" (example only) of npc at 10:59:50, so he will wait for the machine for 10 secs
![wowscrnshot_012717_130323](https://cloud.githubusercontent.com/assets/6797611/22368795/25dd71b4-e491-11e6-8abb-d4e00808d1c4.jpg)

**Comment by Rushor** [comment]
yeah we could do it like that (spawn the Dwarf Magus or Iron Dwarf Assailant some seconds before the machine handles the emerge7submerge emote) thansk for the idea !

**Comment by Teridan-dal-wow** [comment]
Note: The Dwarf Magus or Iron Dwarf Assailant spawned from this script should not give any exp if its killed by player (as they will probably spawn often)
Note 2: Dwarf Magus or Iron Dwarf Assailant don't give any loot at the moment on the server (which is correct), so the ones spawned by the script should also not give any loot.

**Comment by Rushor** [comment]
btw @Teridan-dal-wow are you finshed with stormpeaks? or do you think there are some more locations in stormpeaks that need some scriptlove?

**Comment by Teridan-dal-wow** [comment]
There are few caves left, 2 events: battle between Brunnhildar village and sons of hodir || Iron Dwarf Magus and Iron Dwarf Assailant Part 2 which is a big battle also, and thats it.
But i can leave these 2 big events as they will need ALOT of work to look like retail, what do you have in mind for next?

**Comment by Rushor** [comment]
some few areas i saw after i rechecked stormpeaks

- Branns Base Camp
- Howling Hollow spawns
- pathing in that cave for those creatures http://www.wowhead.com/npc=29409

And yeah would be fine if you post the main idea of the fights for those "big" events first (here if you want), then i could look and decide if it's scriptable with the current state of core and db. thank you!

**Comment by Teridan-dal-wow** [comment]
I will finish the area that i started today and will check these spots you mentioned (maybe tonight or tomorrow). Regarding the "big" events i quickly record an overview of time that will give you basic idea of whats happening.

**Comment by Teridan-dal-wow** [comment]
Here is the overview of the 2 big events left in storm peaks:
https://www.youtube.com/watch?v=JOX1uzGKnhc

**Comment by Rushor** [comment]
the big events will be really hard especially the movement, if the creature was infight once it will be hard to enable the pathmovement, on retail everything looks so fluently - will be really hard. maybe we should work on this after all other stormpeaks stuff is done

FYI: I cannot do anything tomorrow because of a move of my homeplace. on sunday i can fix stuff again :(

**Comment by Teridan-dal-wow** [comment]
As of today, i think all issues in storm peaks with creatures, have been reported (if i missed some or you have any questions let me know).
Regarding these 2 "big" events, i would suggest to leave at the end of northrend (as a very last thing to do),  also i can add a 3-rd one that i forgot about here:
![wowscrnshot_012917_122930](https://cloud.githubusercontent.com/assets/6797611/22403420/2f31483c-e61f-11e6-90e8-661782995199.jpg)
![wowscrnshot_012917_122917](https://cloud.githubusercontent.com/assets/6797611/22403423/2f3679c4-e61f-11e6-82e8-452cb6dd0482.jpg)
![wowscrnshot_012917_122909](https://cloud.githubusercontent.com/assets/6797611/22403421/2f35d604-e61f-11e6-9b10-300885a53fcb.jpg)
![wowscrnshot_012917_122940](https://cloud.githubusercontent.com/assets/6797611/22403422/2f3625f0-e61f-11e6-9dea-49c627410839.jpg)

**Comment by Rushor** [comment]
alright, i will hurry to fix all stuff from stormpeaks now :)
